cloudant-http/src/main/java/com/cloudant/http/HttpConnection.java

```diff
@@ -109,6 +111,8 @@
     public HttpUrlConnectionFactory connectionFactory = new DefaultHttpUrlConnectionFactory();
 
     private int numberOfRetries = 10;
+    private int attemptNumber = 1;
```

Comment:
Sort of. `numberOfRetries` is configurable and counts down as attempts are made. If we don't use `attemptNumber` as well then we'd have to store whatever the initial number was to do a subtraction to work out what number attempt we were on. We could iterate based on `attemptNumber`, but we'd still need a field for the max number to try.

As the primary purpose was to allow correlation of requests and responses I could get rid of it by making the log prefix something like hashcode-numberOfRetries so requests would get logged as 1234abcd-10, then 1234abcd-9 etc.